core/search/term_query.hpp

```diff
@@ -23,54 +23,31 @@
 #ifndef IRESEARCH_TERM_QUERY_H
 #define IRESEARCH_TERM_QUERY_H
 
-#include "filter.hpp"
-#include "cost.hpp"
-#include "utils/string.hpp"
+#include "search/filter.hpp"
 
 NS_ROOT
 
-struct term_reader;
-struct filter_visitor;
-
-//////////////////////////////////////////////////////////////////////////////
-/// @class term_state
-/// @brief cached per reader term state
-//////////////////////////////////////////////////////////////////////////////
-struct reader_term_state {
-  reader_term_state() = default;
-
-  reader_term_state(reader_term_state&& rhs) noexcept
-    : reader(rhs.reader),
-      cookie(std::move(rhs.cookie)) {
-    rhs.reader = nullptr;
-  }
-
-  const term_reader* reader{};
-  seek_term_iterator::cookie_ptr cookie;
-}; // term_state
-
-//////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
 /// @class range_query
 /// @brief compiled query suitable for filters with a single term like "by_term"
-//////////////////////////////////////////////////////////////////////////////
-class term_query : public filter::prepared {
+////////////////////////////////////////////////////////////////////////////////
+class term_query final : public filter::prepared {
  public:
-  typedef states_cache<reader_term_state> states_t;
+  //////////////////////////////////////////////////////////////////////////////
+  /// @class term_state
+  /// @brief cached per reader term state
+  //////////////////////////////////////////////////////////////////////////////
+  struct term_state {
+    term_state() = default;
+    term_state(term_state&& rhs) = default;
```

Comment:
Did you delete copy constructor and assignment operators intentionally?

Reply:
in fact it's implicitly deleted by `std::unique_ptr<...>` (see `seek_term_iterator::cookie_ptr cookie`) member

Reply:
Compare with `multiterm_state::term_state`. You have left `term_state& operator=(term_state&& rhs) = default;`